### PR TITLE
waypoint: fix service matching

### PR DESCRIPTION
**Pull request by dddddai**
**Please provide a description of this PR:**
Services without LabelSelectors match nothing. We are building incorrect waypoint listeners/clusters for these services(which are heavily used in Knative)

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
Good catch, can you add some ut cases to gurantte no future change

File: pilot/pkg/networking/core/v1alpha3/waypoint.go [hunk 1]
```diff
@@ -38,54 +38,29 @@ const (
 	ConnectUpgradeType = "CONNECT"
 )
 
-type WorkloadAndServices struct {
-	WorkloadInfo *model.WorkloadInfo
-	Services     []*model.Service
```
Comment (dddddai) [inline pilot/pkg/networking/core/v1alpha3/waypoint.go:43]: This field is not used, removed for test simplicity

**Review by dddddai — COMMENTED** [review]
Added, thanks

File: pilot/pkg/networking/core/v1alpha3/waypoint.go [hunk 2]
```diff
@@ -38,15 +38,14 @@ const (
...
-}
-
-func findWaypointServices(node *model.Proxy, push *model.PushContext) map[host.Name]*model.Service {
+// findWaypointResources returns workloads and services associated with the waypoint proxy
+func findWaypointResources(node *model.Proxy, push *model.PushContext) ([]*model.WorkloadInfo, map[host.Name]*model.Service) {
```
[the remark this answers is not in the document]
Comment (dddddai) [reply pilot/pkg/networking/core/v1alpha3/waypoint.go:42]: I think the idea is to call `push.WorkloadsForWaypoint` only once to fetch both workloads and services, which are used to [build waypoint listeners](https://github.com/istio/istio/blob/bd79a2cf2b9b335a0b576afe4ffa6b2f8bae1cee/pilot/pkg/networking/core/v1alpha3/listener_waypoint.go#L65)

If we split it up we will have to call it twice

**Review by howardjohn — APPROVED** [review]

**Pull request merged**